src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -815,34 +816,82 @@ internal static int GetHashCodeOrdinalIgnoreCase(ReadOnlySpan<char> value)
         // or are otherwise mitigated
         internal unsafe int GetNonRandomizedHashCode()
         {
+            uint hash1 = (5381 << 16) + 5381;
+            uint hash2 = hash1;
+
             fixed (char* src = &_firstChar)
             {
                 Debug.Assert(src[this.Length] == '\0', "src[this.Length] == '\\0'");
-                Debug.Assert(((int)src) % 4 == 0, "Managed string should start at 4 bytes boundary");
-
-                uint hash1 = (5381 << 16) + 5381;
-                uint hash2 = hash1;
+                Debug.Assert(((int) src) % 4 == 0, "Managed string should start at 4 bytes boundary");
 
-                uint* ptr = (uint*)src;
+                uint* ptr = (uint*) src;
                 int length = this.Length;
 
+                // We "normalize to lowercase" every char by ORing with 0x0020. This casts
```

Comment:
This comment is misplaced. Copy&paste bug?

Reply:
Yes, will clean it up, thanks for pointing out.